python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -475,51 +475,33 @@ def accessURL(self, thisSE=''):
         if len(self.subfiles) == 0:
 
             files_URLs = self._remoteURLs
-            this_accessURL = ''
+            this_accessURL = {}
 
             for this_SE in files_URLs.keys():
-
-                this_URL = files_URLs.get(this_SE)
-                these_sites_output = execute('getSitesForSE("%s")' % str(this_SE))
-                if thisSE == '':
-                    default_site = execute('getSiteForSE("%s")' % self.defaultSE)
-                else:
-                    default_site = thisSE
-
-                if these_sites_output.get('OK', False):
-                    these_sites = these_sites_output.get('Value')
-                    for this_site in these_sites:
-                        if type(default_site) == type([]):
-                            hasMatched = False
-                            for this_Site_in_SE in default_site:
-                                if this_site == this_Site_in_SE:
-                                    hasMatched = True
-                                    break
-                            if hasMatched:
-                                break
-                        elif type(default_site) == type(''):
-                            if this_site == default_site:
-                                this_accessURL = this_URL
-                                break
-                if this_accessURL != '':
-                    break
-
-            # Cannot find an accessURL
+                this_accessURL[this_SE] = []
+                myurl = execute('getAccessURL("%s" , "%s")' % (self.lfn , str(this_SE)))
```

Comment:
There should be no space between `self.lfn` and the `,`. Also, there is no need to pass `this_SE` via `str()`